packages/application-generic/src/usecases/digest-filter-steps/digest-filter-steps-timed.usecase.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@nestjs/common';
+import {
+  JobStatusEnum,
+  JobRepository,
+  NotificationStepEntity,
+  NotificationRepository,
+} from '@novu/dal';
+import {
+  IDigestTimedMetadata,
+  StepTypeEnum,
+  DigestTypeEnum,
+} from '@novu/shared';
+
+import { DigestFilterStepsCommand } from './digest-filter-steps.command';
+import { DigestFilterSteps } from './digest-filter-steps.usecase';
+
+@Injectable()
+export class DigestFilterStepsTimed {
```

Comment:
the use-case to filter steps for the timed digest run... basically at first trigger we will create all the jobs from the template steps; on subsequent triggers if digest is running we will only create the jobs from the steps that are before the digest

Reply:
How is this different from the regular backoff?

Reply:
Not much but I've improved it a little. 

**The regular digest filter use-case:** always creates a jobs array based on template steps, even if there is a digest job delayed, and then when the digest is running it marks all other digest jobs as completed. This leads to other jobs in the pending state forever.

**The timed digest filter use-case:** obviously creates jobs based on the template steps, but it checks if there is a **delayed timed digest job**, and if there is it just creates a jobs array from the steps that are before the digest, so like it cuts the jobs array.

I've asked this question before on the Discord, but no one has answered it: https://discord.com/channels/895029566685462578/1045289346598703164/1096406448839331941